### PR TITLE
DBZ-7138 Additional Conditions for MongoDb Incremental Snapshot

### PR DESCRIPTION
**Pull request by ani-sha**
- [x] Add documentation - https://github.com/debezium/debezium/pull/5504

closes https://issues.redhat.com/browse/DBZ-7138

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -589,13 +623,22 @@ private Document constructQueryPredicate(Object[] startKey, Object[] endKey) {
...
+                predicate = new Document("$and", Arrays.asList(chunkEndPredicate, maxKeyPredicate));
+            }
         }
-
         return predicate;
     }
 
+    private String getAdditionalConditions() {
+        // Strip additional parenthesis to make sure additional conditions are parsed correctly
+        return context.currentDataCollectionId().getAdditionalCondition().map(s -> s.substring(1, s.length() - 1)).orElse("");
```
Comment (mfvitale) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/snapshot/MongoDbIncrementalSnapshotChangeEventSource.java:639]: WDYT to put ```s -> s.substring(1, s.length() - 1)``` in a method with a meaningful name?

**Comment by mfvitale** [comment]
@ani-sha LGTM, I leaved just a minor comment.

**Review by jcechace — CHANGES_REQUESTED** [review]
Small nitpick as pointed out by @mfvitale

**Comment by jcechace** [comment]
@jpechane  should we also backport this to 2.6?

**Comment by jpechane** [comment]
@jcechace No, this is technically new feature so there will be no backport

**Pull request merged**